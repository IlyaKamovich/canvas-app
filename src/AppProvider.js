import React, { useReducer, useContext, createContext } from 'react';

const noop = () => {};

const StateContext = createContext(noop);
const DispatchContext = createContext(noop);

export const AppActions = {
  PROCESS_FILE_SUCCESS: 'PROCESS_FILE_SUCCESS',
  PROCESS_FILE_ERROR: 'PROCESS_FILE_ERROR',
  LOADING: 'LOADING',
};

function appReducer(state, action) {
  switch (action.type) {
    case AppActions.PROCESS_FILE_SUCCESS:
      const { result, url } = action.payload;
      return { ...state, result, url };
    case AppActions.PROCESS_FILE_ERROR:
      return { ...state, error: action.payload };
    default:
      return state;
  }
}

function AppProvider({ children }) {
  const [state, dispatch] = useReducer(appReducer, {});

  return (
    <StateContext.Provider value={state}>
      <DispatchContext.Provider value={dispatch}>
        {children}
      </DispatchContext.Provider>
    </StateContext.Provider>
  );
}

export function useAppState() {
  const context = useContext(StateContext);
  if (context === undefined) {
    throw new Error('useAppState must be used within a AppProvider');
  }
  return context;
}

export function useAppDispatch() {
  const context = useContext(DispatchContext);
  if (context === undefined) {
    throw new Error('useAppDispatch must be used within a AppProvider');
  }
  return context;
}

export default AppProvider;
