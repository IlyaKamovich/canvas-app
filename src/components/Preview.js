import React from 'react';

import './Preview.css';

function Preview(props) {
  return (
    <div className="preview">
      <textarea />
    </div>
  );
}

export default Preview;
