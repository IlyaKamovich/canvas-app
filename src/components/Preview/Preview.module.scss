@value inputDataFieldWidth from '../DateFileInput/DataFileInput.module.scss';

.preview__container {
  position: relative;
  margin-top: 3rem;
  width: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}

.preview__area {
  width: inputDataFieldWidth;
  height: 50rem;
  border: none;
  box-shadow: 0 2rem 6rem 1rem #747681;
  padding: 5px;
  font-family: Tahoma, sans-serif;
  resize: none;
  background-color: #d5d5d5e0;
  cursor: initial;
}
