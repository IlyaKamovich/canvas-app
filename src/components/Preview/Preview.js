import React from 'react';

import styles from './Preview.module.scss';
import Spinner from '../Spinner/Spinner';
import { useAppState } from '../../AppProvider';

const spinnerPosition = {
  position: 'absolute',
  top: '50%',
  left: '50%',
  transform: 'translate(-50%, -50%)',
};

const Error = ({ children }) => <p style={{ color: 'red' }}>{children}</p>;

function Preview() {
  const state = useAppState();

  const { loading = false, snapshot = 'There is no data yet', error } = state;

  return (
    <div className={styles.preview__container}>
      {loading && <Spinner style={spinnerPosition} />}
      <div className={styles.preview__area}>
        {error ? <Error>{error}</Error> : <pre>{snapshot}</pre>}
      </div>
    </div>
  );
}

export default Preview;
