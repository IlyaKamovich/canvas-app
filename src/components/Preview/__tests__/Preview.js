import React from 'react';
import { render } from '@testing-library/react';
import Preview from '../Preview';
import AppProvider from '../../../AppProvider';

describe('<Preview /> spec', () => {
  it('renders the component', () => {
    const { container } = render(
      <AppProvider>
        <Preview />
      </AppProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
