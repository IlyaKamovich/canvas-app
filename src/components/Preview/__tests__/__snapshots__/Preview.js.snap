// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Preview /> spec renders the component 1`] = `
<div>
  <div
    class="preview__container"
  >
    <div
      class="preview__area"
    >
      <pre>
        There is no data yet
      </pre>
    </div>
  </div>
</div>
`;
